src/main/rx/crsf.c

```diff
@@ -175,6 +180,40 @@ static void handleCrsfLinkStatisticsFrame(const crsfLinkStatistics_t* statsPtr,
     }
 #endif
 
+#ifdef USE_RX_LINK_UPLINK_POWER
+    uint16_t crsfPowerInMw;
+    switch (stats.uplink_TX_Power) {
+    case 0:
+        crsfPowerInMw = 0;
+        break;
+    case 1:
+        crsfPowerInMw = 10;
+        break;
+    case 2:
+        crsfPowerInMw = 25;
+        break;
+    case 3:
+        crsfPowerInMw = 100;
+        break;
+    case 4:
+        crsfPowerInMw = 500;
+        break;
+    case 5:
+        crsfPowerInMw = 1000;
+        break;
+    case 6:
+        crsfPowerInMw = 2000;
+        break;
+    case 7:
+        crsfPowerInMw = 250;
+        break;
+    default:
+        crsfPowerInMw = 0;
+        break;
+    }
+    rxSetUplinkTxPwr(crsfPowerInMw);
+#endif
```

Comment:
Doing it this way is more code and also means that `uplinkTXPowerStates[]` is no longer required. A better way would be to define the number of states and then validate against that count. For example where you define the power states array:
```
#define CRSF_UPLINK_POWER_LEVEL_COUNT 8
const uint16_t uplinkTXPowerStates[CRSF_UPLINK_POWER_LEVEL_COUNT] = {0, 10, 25, 100, 500, 1000, 2000, 250};
```
And then here try this:
```
#ifdef USE_RX_LINK_UPLINK_POWER
    const uint8_t crsfUplinkPowerIndex = (stats.uplink_TX_Power < CRSF_UPLINK_POWER_LEVEL_COUNT) ? stats.uplink_TX_Power : 0;
    rxSetUplinkTxPwr(uplinkTXPowerStates[crsfUplinkPowerIndex]);
#endif
```

Reply:
Thank you for this. Done. Tested it and it works as expected.